base/applications/osk/main.c

```diff
@@ -822,32 +1008,50 @@ int WINAPI wWinMain(HINSTANCE hInstance,
     /* Load the application's settings from the registry */
     LoadSettings();
 
-    /* If the member of the struct (bShowWarning) is set then display the dialog box */
-    if (Globals.bShowWarning)
-    {
-        /* If for whatever reason the thread fails to be created then handle the dialog box in main thread... */
-        if (CreateThread(NULL, 0, OSK_WarningDlgThread, (PVOID)Globals.hInstance, 0, NULL) == NULL)
-        {
-            DialogBoxW(Globals.hInstance, MAKEINTRESOURCEW(IDD_WARNINGDIALOG_OSK), Globals.hMainWnd, OSK_WarningProc);
-        }
-    }
-
-    /* Before initializing the dialog execution, check if the chosen keyboard type is standard or enhanced */
-    if (Globals.bIsEnhancedKeyboard)
-    {
-        LayoutResource = MAIN_DIALOG_ENHANCED_KB;
-    }
-    else
+    /* Define the window class */
+    wc.cbSize        = sizeof(wc);
+    wc.hInstance     = Globals.hInstance;
+    wc.lpfnWndProc   = OSK_WndProc;
+    wc.lpszMenuName  = MAKEINTRESOURCEW(IDR_OSK_MENU);
+    wc.lpszClassName = OSK_CLASS;
+    wc.style         = CS_HREDRAW | CS_VREDRAW;
+    wc.hbrBackground = (HBRUSH)(COLOR_WINDOW + 1);
+    /* Set the application's icon */
+    wc.hIcon = LoadImageW(Globals.hInstance, MAKEINTRESOURCEW(IDI_OSK), IMAGE_ICON, 0, 0, LR_SHARED | LR_DEFAULTSIZE);
+    wc.hIconSm = CopyImage(wc.hIcon, IMAGE_ICON, GetSystemMetrics(SM_CXSMICON), GetSystemMetrics(SM_CYSMICON), LR_COPYFROMRESOURCE);
+
+    if (!RegisterClassExW(&wc))
+        goto quit;
+
+    if (!LoadStringW(Globals.hInstance, STRING_OSK, Globals.szTitle, _countof(Globals.szTitle)))
+        StringCchCopyW(Globals.szTitle, _countof(Globals.szTitle), L"On-Screen Keyboard");
```

Comment:
I am not keen of hard-coding the window title only in English locale when there's `STRING_OSK` for each translation file instead. Please use that.

Reply:
it's fallback if `LoadStringW` fails to load the relevant string from the resources

Reply:
Aaaaf my eyes failed hard on this one, my bad lol. Still, the possibility for `LoadStringW` to fail loading the string title from resource is slim to none. So I think having this fallback is quite redundant (unless you provide a reasonable explanation as of how this function cal fail, and if such failure rate is higher than it should be).

Reply:
At some point people will add failure checks for stuff like `2+2` ...